src/test/scala/org/camunda/feel/impl/builtin/BuiltinContextFunctionsTest.scala

```diff
@@ -62,4 +63,116 @@ class BuiltinContextFunctionsTest
     eval(""" get value({}, "foo") """) should be(ValNull)
   }
 
+  "A put function" should "add an entry to an empty context" in {
+
+    eval(""" put({}, "x", 1) """) should be(
+      ValContext(
+        StaticContext(variables = Map("x" -> ValNumber(1)))
+      ))
+  }
+
+  it should "add an entry to an existing context" in {
+
+    eval(""" put({x:1}, "y", 2) """) should be(
+      ValContext(
+        StaticContext(variables = Map("x" -> ValNumber(1), "y" -> ValNumber(2)))
+      ))
+  }
+
+  it should "override an entry of an existing context" in {
+
+    eval(""" put({x:1}, "x", 2) """) should be(
+      ValContext(
+        StaticContext(variables = Map("x" -> ValNumber(2)))
+      ))
+  }
+
+  it should "add a context entry to an existing context" in {
+
+    eval(""" put({x:1}, "y", {"z":2}) = {x:1, y:{z:2} } """) should be(
+      ValBoolean(true))
+  }
+
+  it should "return null if the value is not present" in {
+
+    eval(""" put({}, "x", notExisting) """) should be(ValNull)
+  }
+
+  "A put all function" should "return a single empty context" in {
+
+    eval(""" put all({}) """) should be(
+      ValContext(
+        StaticContext(variables = Map.empty)
+      ))
+  }
+
+  it should "return a single context" in {
+
+    eval(""" put all({x:1}) """) should be(
+      ValContext(
+        StaticContext(variables = Map("x" -> ValNumber(1)))
+      ))
+  }
+
+  it should "combine empty contexts" in {
+
+    eval(""" put all({}, {}) """) should be(
+      ValContext(
+        StaticContext(variables = Map.empty)
+      ))
+  }
+
+  it should "add all entries to an empty context" in {
+
+    eval(""" put all({}, {x:1}) """) should be(
+      ValContext(
+        StaticContext(variables = Map("x" -> ValNumber(1)))
+      ))
+  }
+
+  it should "add an entry to an context" in {
+
+    eval(""" put all({x:1}, {y:2}) """) should be(
+      ValContext(
+        StaticContext(variables = Map("x" -> ValNumber(1), "y" -> ValNumber(2)))
+      ))
+  }
+
+  it should "add all entries to an context" in {
+
+    eval(""" put all({x:1}, {y:2, z:3}) """) should be(
+      ValContext(
+        StaticContext(variables =
+          Map("x" -> ValNumber(1), "y" -> ValNumber(2), "z" -> ValNumber(3)))
+      ))
+  }
+
+  it should "override an entry of the existing context" in {
+
+    eval(""" put all({x:1}, {x:2}) """) should be(
+      ValContext(
+        StaticContext(variables = Map("x" -> ValNumber(2)))
+      ))
+  }
+
+  it should "combine three contexts" in {
+
+    eval(""" put all({x:1}, {y:2}, {z:3}) """) should be(
+      ValContext(
+        StaticContext(variables =
+          Map("x" -> ValNumber(1), "y" -> ValNumber(2), "z" -> ValNumber(3)))
+      ))
+  }
+
+  it should "add a nested context" in {
+
+    eval(""" put all({x:1}, {y:{z:2}}) = {x:1, y:{z:2} } """) should be(
```

Comment:
Why the predicate here instead of checking the entries like in all other tests?

Reply:
The nested structure is harder to test. The inner `context` is an interface with different implementations. 
We could extract the variables out of it in the test case. But it is easier to read if we check it in the expression itself.